### ISSUE TITLE
This is completely wrong, you are using val set and test set for training.

### ISSUE THREAD

**Issue by juck2018** [xiangdal/TrajectoryNet#1]
Please review the code here, the perm index covers the test_index and valid index. You are basically overfitting the val set and test set.

`    perm = np.random.permutation(X_train.shape[1])    
    X_train = X_all[:,perm,:]
    y_train = Y_all[perm,:]
    stop_train = e_stop_all[perm]

    X_test = X_all[:,test_index,:]
    y_test = Y_all[test_index,:]
    stop_test = e_stop_all[test_index]

    X_valid = X_all[:,valid_index,:]
    y_valid = Y_all[valid_index,:]
    stop_valid = e_stop_all[valid_index]`

**Comment by xiangdal** [comment]
Hi @juck2018 ,

Thank you for your comment! It is indeed a bug that was introduced while I try to open source the code. The following code should be deleted because the training data was already permuted in `np.random.shuffle(train_index)`. I will make another commit to update the model and results.

`X_test = X_all[:,test_index,:]
y_test = Y_all[test_index,:]
stop_test = e_stop_all[test_index]`
